Task 'buildNativeLambda' not found. Works with `-Pmicronaut.runtime=lambda_provided`

### Expected Behavior

In a multi project build, I expected to be able to run:

`./gradlew :app:buildNativeLambda`

It works if I do: 

`./gradlew :app:buildNativeLambda -Pmicronaut.runtime=lambda_provided`


### Actual Behaviour

```
./gradlew :app:buildNativeLambda                                     

FAILURE: Build failed with an exception.

* What went wrong:
Task 'buildNativeLambda' not found in project ':app'.

* Try:
> Run gradlew tasks to get a list of available tasks.
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.

* Get more help at https://help.gradle.org

```

### Steps To Reproduce

Clone: 

https://github.com/grails-core-issues-forks/micronaut-gradle-plugin-buildNativeLambda

Run `./gradlew :app:buildNativeLambda`

If you run `./gradlew :app:buildNativeLambda -Pmicronaut.runtime=lambda_provided `it works. 

### Environment Information

_No response_

### Example Application

https://github.com/grails-core-issues-forks/micronaut-gradle-plugin-buildNativeLambda

### Version

3.5.0-SNAPSHOT

--- a/starter-core/src/main/java/io/micronaut/starter/feature/build/gradle/Gradle.java
+++ b/starter-core/src/main/java/io/micronaut/starter/feature/build/gradle/Gradle.java
@@ -15,13 +15,16 @@
  */
 package io.micronaut.starter.feature.build.gradle;
 
+import io.micronaut.core.annotation.NonNull;
 import io.micronaut.starter.application.ApplicationType;
 import io.micronaut.starter.application.generator.GeneratorContext;
+import io.micronaut.starter.build.Property;
 import io.micronaut.starter.build.gradle.GradleBuild;
 import io.micronaut.starter.build.gradle.GradleBuildCreator;
 import io.micronaut.starter.build.gradle.GradlePlugin;
 import io.micronaut.starter.feature.Feature;
 import io.micronaut.starter.feature.FeatureContext;
+import io.micronaut.starter.feature.MicronautRuntimeFeature;
 import io.micronaut.starter.feature.build.BuildFeature;
 import io.micronaut.starter.feature.build.KotlinBuildPlugins;
 import io.micronaut.starter.feature.build.MicronautBuildPlugin;
@@ -37,7 +40,9 @@ import io.micronaut.starter.template.Template;
 import io.micronaut.starter.template.URLTemplate;
 import jakarta.inject.Singleton;
 
+import java.util.List;
 import java.util.Set;
+import java.util.stream.Collectors;
 
 import static io.micronaut.starter.build.Repository.micronautRepositories;
 
@@ -95,10 +100,17 @@ public class Gradle implements BuildFeature {
         )));
 
         generatorContext.addTemplate("gitignore", new RockerTemplate(Template.ROOT, ".gitignore", gitignore.template()));
-        generatorContext.addTemplate("projectProperties", new RockerTemplate(Template.ROOT, "gradle.properties", gradleProperties.template(generatorContext.getBuildProperties().getProperties())));
+        generatorContext.addTemplate("projectProperties", new RockerTemplate(Template.ROOT, "gradle.properties", gradleProperties.template(gradleProperties(generatorContext))));
 
         String settingsFile = buildTool == BuildTool.GRADLE ? "settings.gradle" : "settings.gradle.kts";
         generatorContext.addTemplate("gradleSettings", new RockerTemplate(Template.ROOT, settingsFile, settingsGradle.template(generatorContext.getProject(), build, generatorContext.getModuleNames())));
+    }
+
+    @NonNull
+    private static List<Property> gradleProperties(@NonNull GeneratorContext generatorContext) {
+        return generatorContext.getBuildProperties().getProperties().stream()
+                .filter(p -> p.getKey() == null || !p.getKey().equals(MicronautRuntimeFeature.PROPERTY_MICRONAUT_RUNTIME)) // It is set via the DSL
+                .collect(Collectors.toList());
     }
 
     @Override

--- a/starter-core/src/test/groovy/io/micronaut/starter/feature/function/awslambda/AwsLambdaSpec.groovy
+++ b/starter-core/src/test/groovy/io/micronaut/starter/feature/function/awslambda/AwsLambdaSpec.groovy
@@ -3,7 +3,9 @@ package io.micronaut.starter.feature.function.awslambda
 import io.micronaut.starter.ApplicationContextSpec
 import io.micronaut.starter.BuildBuilder
 import io.micronaut.starter.application.ApplicationType
+import io.micronaut.starter.feature.MicronautRuntimeFeature
 import io.micronaut.starter.feature.build.gradle.Gradle
+import io.micronaut.starter.feature.coherence.CoherenceFeature
 import io.micronaut.starter.fixture.CommandOutputFixture
 import io.micronaut.starter.options.*
 import spock.lang.Shared
@@ -15,6 +17,17 @@ class AwsLambdaSpec extends ApplicationContextSpec implements CommandOutputFixtu
     @Shared
     @Subject
     AwsLambda awsLambda = beanContext.getBean(AwsLambda)
+
+    void 'test gradle.properties does not contain micronaut.runtime'() {
+        when:
+        Map<String, String> output = generate(['aws-lambda'])
+        String properties = output["gradle.properties"]
+
+        then:
+        properties
+        !properties.contains(MicronautRuntimeFeature.PROPERTY_MICRONAUT_RUNTIME)
+        output["build.gradle"].contains('runtime("lambda_java")')
+    }
 
     void 'test readme.md with feature aws-lambda contains links to micronaut docs'() {
         when:

